### PR TITLE
Update type representations and new REST endpoints

### PR DESCRIPTION
**Pull request by fppt**

### CODE REVIEW

File: grakn-engine/src/main/java/ai/grakn/engine/controller/response/ConceptBuilder.java [hunk 1]
```diff
@@ -80,10 +80,10 @@ public static Things buildThings(ai.grakn.concept.Type type, int offset, int lim
     //TODO: This will scale poorly with super nodes. Need to introduce some sort of paging maybe?
```
Comment (aelred) [inline grakn-engine/src/main/java/ai/grakn/engine/controller/response/ConceptBuilder.java]: I think we just did the TODO!

File: grakn-engine/src/main/java/ai/grakn/engine/controller/response/MetaConcept.java [hunk 2]
```diff
@@ -46,9 +44,9 @@ public static MetaConcept create(
             @JsonProperty("id") ConceptId id,
             @JsonProperty("@id") Link selfLink,
             @JsonProperty("label") Label label,
-            @JsonProperty("super") @Nullable Link sup,
-            @JsonProperty("subs") Set<Link> subs
+            @JsonProperty("super") @Nullable EmbeddedSchemaConcept sup,
+            @JsonProperty("subs") Link subs
     ){
-        return new AutoValue_MetaConcept(Schema.BaseType.TYPE.name(), id, selfLink, label, false, sup, subs, true, Collections.emptySet(), Collections.emptySet(), Collections.emptySet());
+        return new AutoValue_MetaConcept(Schema.BaseType.TYPE.name(), id, selfLink, label, false, sup, subs, true, null, null, null, null);
```
Comment (aelred) [inline grakn-engine/src/main/java/ai/grakn/engine/controller/response/MetaConcept.java]: These could be regular links to empty collections if we want. That feels like a bit less of a special-case to me.

File: grakn-engine/src/main/java/ai/grakn/engine/controller/response/ConceptBuilder.java (hunk 1 above)
Comment (fppt) [reply grakn-engine/src/main/java/ai/grakn/engine/controller/response/ConceptBuilder.java]: Our paging is artificial I think, so i will keep the todo for the moment.

Comment (aelred) [reply grakn-engine/src/main/java/ai/grakn/engine/controller/response/ConceptBuilder.java]: oh yeah! I forgot about that.

File: grakn-engine/src/main/java/ai/grakn/engine/controller/response/MetaConcept.java (hunk 2 above)
Comment (fppt) [reply grakn-engine/src/main/java/ai/grakn/engine/controller/response/MetaConcept.java]: Ya fair enough. Less nulls running around.

**Comment by aelred** [comment]
@fppt you need to merge with `stable`

**Comment by fppt** [comment]
@aelred thanks for the heads up!

**Pull request merged**